packages/adblocker-electron-preload/src/index.ts

```diff
@@ -8,25 +8,22 @@
 
 import { ipcRenderer } from 'electron';
 
-import {
-  DOMMonitor,
-  IBackgroundCallback,
-  IMessageFromBackground,
-  injectScript,
-} from '@ghostery/adblocker-content';
+import { DOMMonitor, IBackgroundCallback } from '@ghostery/adblocker-content';
 
-function getCosmeticsFiltersFirst(): string[] | null {
-  return ipcRenderer.sendSync('get-cosmetic-filters-first', window.location.href);
-}
-function getCosmeticsFiltersUpdate(data: Omit<IBackgroundCallback, 'lifecycle'>) {
-  ipcRenderer.send('get-cosmetic-filters', window.location.href, data);
+function injectCosmeticFilters(data?: Omit<IBackgroundCallback, 'lifecycle'>): Promise<void> {
+  return ipcRenderer.invoke(
+    '@ghostery/adblocker/inject-cosmetic-filters',
+    window.location.href,
+    data,
+  );
 }
 
 if (window === window.top && window.location.href.startsWith('devtools://') === false) {
-  (() => {
-    const enableMutationObserver = ipcRenderer.sendSync('is-mutation-observer-enabled');
+  (async () => {
+    const enableMutationObserver = await ipcRenderer.invoke(
```

Comment:
I don't think this is needed. It should be fine to always create the DOM_MONITOR but only start it when needed. With this line removed, the wrapping function can be sync again